src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
```

Comment:
- Constant variables should be called in UPPER_CASE. 
- You should write the whole name (default), but no short versions like 'def', 'arr' and so on.
- There is no need of creation 'empty_arraydata' variable, because you use it only once and you can replace that usage by just an empty array, so, you should remove this variable.

You can read more about naming rules [here](https://mate-academy.github.io/style-guides/java/java.html#s5-naming)
```suggestion
    private static final int DEFAULT_CAPACITY = 10;
```